### PR TITLE
Add project collaboration

### PR DESCRIPTION
**Pull request by Mubangizi**
# Description
Adds Project collaboration and IAM feature for resources under a project
- This PR adds CRUD operations for project users
- Implements access management to projects and resources in them

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Trello Ticket ID
Trello ticket [link](https://trello.com/c/oIn95KAb/1219-implement-project-collaboration-in-the-backend)

## How Can This Be Tested?
Checkout the following endpoints
- POST - `/projects/{project_id}/users` - to add User to project
- GET - `/projects/{project_id}/users` - to get Users in a project
- PATCH - `/projects/{project_id}/users` - to update user role in a project
- DELETE - `/projects/{project_id}/users` -  to remove User from a project
- POST - `/projects/{project_id}/users/transfer` - to transfer project ownership to another User

### CODE REVIEW

File: api_docs.yml [hunk 1]
```diff
@@ -2572,6 +2572,182 @@ paths:
         500:
           description: "Internal Server Error"
 
+  "/projects/{project_id}/users":
+    post:
+      tags:
+        - projects
+      consumes:
+        - application/json
+      description: "Add a user to a project, role can be either 'owner' or 'admin' or 'member'"
```
Comment (MutegekiHenry) [inline api_docs.yml:2581]: I think the owner role should only be assigned at project creation

Comment (MutegekiHenry) [inline api_docs.yml:2581]: I think the owner role should only be assigned at project creation

**Review by MutegekiHenry — APPROVED** [review]
Great work on this Allan

File: api_docs.yml (hunk 1 above)
Comment (Mubangizi) [reply api_docs.yml:2581]: Oh yeah , that's just in API docs,, but we don't accept owners

**Review by MutegekiHenry — APPROVED** [review]
LGMT

**Review by e-ian — APPROVED** [review]
Nice work @Mubangizi

**Pull request merged**